src/ServicePulse.Host/vue/src/composables/serviceMonitoringEndpoints.ts

```diff
@@ -0,0 +1,163 @@
+import { useGetExceptionGroups } from "./serviceMessageGroup";
+import { type Endpoint, type GroupedEndpoint, type EndpointGroup, type EndpointDetails, emptyEndpointMetrics } from "@/resources/Endpoint";
+
+/**
+ * @returns the max number of segments in a array of endpoint object names
+ */
+export function useFindEndpointSegments(endpoints: Endpoint[]) {
+  return endpoints.reduce(function (acc, cur) {
+    return Math.max(acc, cur.name.split(".").length - 1);
+  }, 0);
+}
+
+/**
+ * @param historyPeriod - The history period value.  The default is (1)
+ * @returns A array of monitoring endpoint objects
+ */
+export async function useGetAllMonitoredEndpoints(historyPeriod = 1) {
+  let endpoints: Endpoint[] = [];
+  if (!useIsMonitoringDisabled() && !monitoringConnectionState.unableToConnect) {
+    try {
+      const response = await useFetchFromMonitoring(`${`monitored-endpoints`}?history=${historyPeriod}`);
+      const data = response && (await response.json());
+      endpoints = data ?? [];
+      await addEndpointsFromScSubscription(endpoints);
+    } catch (error) {
+      console.error(error);
+    }
+  } else {
+    await addEndpointsFromScSubscription(endpoints);
+  }
+  return endpoints;
+}
+
+/**
+ * @param filterString - The text entered by the user when filtering monitoring endpoints by name
+ * @returns A filtered array of monitoring endpoint objects
+ */
+export function useFilterAllMonitoredEndpointsByName(endpoints: Endpoint[], filterString: string) {
+  if (!filterString) {
+    return endpoints;
+  }
+  return endpoints.filter((endpoint) => endpoint.name.toLowerCase().includes(filterString.toLowerCase()));
+}
+
+/**
+ * @param endpoints - An array of endpoint objects from ServiceControl
+ * @param numberOfSegments - The number of segments to group the array of endpoints by
+ * @returns An array of grouped endpoint objects
+ */
+export function useGroupEndpoints(endpoints: Endpoint[], numberOfSegments: number): EndpointGroup[] {
+  const groups = new Map<string, EndpointGroup>();
+  for (const element of endpoints) {
+    const grouping = parseEndpoint(element, numberOfSegments);
+
+    const resultGroup = groups.get(grouping.groupName) ?? {
+      group: grouping.groupName,
+      endpoints: [],
+    };
+    resultGroup.endpoints.push(grouping);
+    groups.set(grouping.groupName, resultGroup);
+  }
+  return [...groups.values()];
+}
+
+/**
+ * @param endpointName - the endpoint name whose details is needed
+ * @param The history period value.  The default is (1)
+ * @returns The details of the endpoint
+ */
+export function useGetEndpointDetails(endpointName: string, historyPeriod = 1) {
+  const data = ref<EndpointDetails | null>(null);
+  return {
+    data,
+    refresh: async () => {
+      if (!useIsMonitoringDisabled() && !monitoringConnectionState.unableToConnect) {
+        try {
+          const response = await useFetchFromMonitoring(`${`monitored-endpoints`}/${endpointName}?history=${historyPeriod}`);
```

Comment:
I created a typed API for the main instance, have a look at https://github.com/Particular/ServicePulse/blob/41fb564ab1e17b28a288170f575be1a0910f4281/src/ServicePulse.Host/vue/src/composables/serviceServiceControlUrls.ts#L18, I think you could do the same here, thoughts?